yt/funcs.py

```diff
@@ -308,22 +318,72 @@ def insert_ipython(num_up=1):
 #
 
 
-class TqdmProgressBar:
-    # This is a drop in replacement for pbar
-    # called tqdm
-    def __init__(self, title, maxval):
-        self._pbar = tqdm(leave=True, total=maxval, desc=title)
-        self.i = 0
+class CompletionColumn(ProgressColumn):
+    """Renders the current number of iterations completed."""
 
-    def update(self, i=None):
-        if i is None:
-            i = self.i + 1
-        n = i - self.i
-        self.i = i
-        self._pbar.update(n)
+    def render(self, task: Task) -> Text:
+        return Text(f"{task.completed} it", style="progress.percentage")
+
+
+class CompletionSpeedColumn(ProgressColumn):
+    """Renders human readable completion speed."""
+
+    # This is based off rich.progress.TransferSpeedColumn
+
+    def render(self, task: Task) -> Text:
+        """Show data transfer speed."""
+        speed = task.finished_speed or task.speed
+        if speed is None:
+            return Text("?", style="progress.data.speed")
+        data_speed = int(speed)
+        return Text(f"{data_speed} it/s", style="progress.data.speed")
+
+
+class RichProgressBar:
+    """A thin wrapper around rich.progress.Progress"""
+
+    def __init__(
+        self,
+        title: str,
+        maxval: Optional[int] = None,
+        columns: Optional[Sequence[ProgressColumn]] = None,
+    ):
+        if columns is None:
+            columns = [
+                SpinnerColumn(),
+                TextColumn("[progress.description]{task.description}"),
+            ]
+            if maxval is None:
+                columns += [
+                    CompletionColumn(),
+                    TextColumn("[progress.data.speed]@"),
+                    CompletionSpeedColumn(),
+                ]
+                # this is a workaround. As of rich 9.13.0, some of the columns
+                # we're using here require a non-None value in `task.total` to render
+                maxval = np.inf
```

Comment:
no need to use numpy here actually
```suggestion
                maxval = float("inf")
```